internal/jimm/juju/modelbuilder.go

```diff
@@ -18,6 +20,23 @@ import (
 	"github.com/canonical/jimm/v3/internal/openfga"
 )
 
+type candidateController struct {
+	// controller is a candidate controller for the new model.
+	controller dbmodel.Controller
+	// priority decides an ordering for controller selection
+	// based on the cloud-region.
+	priority uint
+}
+
+func shuffleCandidateControllers(controllers []candidateController) {
+	shuffle(len(controllers), func(i, j int) {
+		controllers[i], controllers[j] = controllers[j], controllers[i]
+	})
+	sort.SliceStable(controllers, func(i, j int) bool {
```

Comment:
I noticed this isn't tested, could we add a test where it sets rand.Seed? It is our most important intelligent function of course aha 😝